cluster/addons/rbac/cluster-loadbalancing/glbc/clusterrole.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: system:controller:glbc
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+rules:
+- apiGroups: [""]
+  resources: ["secrets", "endpoints", "services", "pods", "nodes", "namespaces", "configmaps", "events"]
+  verbs: ["get", "list", "watch", "update", "create", "patch"]
```

Comment:
can we separate these into resources glbc needs read access on, and those it needs write access on? I sincerely hope it does not actually need create/update/patch access to all of these resources

Reply:
@rramkumar1 thoughts? The code itself can be a bit opaque

Reply:
@liggitt You are correct that it probably does not actually need all verbs on these resources although I can't remember the reason for doing it this way.

Reply:
added a TODO for the accounting/pruning: https://github.com/kubernetes/ingress-gce/issues/743

Reply:
sweeping through the ingress-gce code:

1. configmap use is limited to the kube-system namespace, so should be granted in a Role, not a ClusterRole:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: system:controller:glbc
  namespace: kube-system
  labels:
    addonmanager.kubernetes.io/mode: Reconcile
rules:
- apiGroups: [""]
  resources: ["configmaps"]
  verbs: ["get", "list", "watch", "update", "create", "patch"]
---
apiVersion: rbac.authorization.k8s.io/
kind: RoleBinding
metadata:
  name: system:controller:glbc
  namespace: kube-system
  labels:
    addonmanager.kubernetes.io/mode: Reconcile
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: system:controller:glbc
subjects:
- kind: User
  name: system:controller:glbc
```

2. endpoints, pods, nodes, namespaces are not written to and can be limited to read-only, and services are not created/deleted, only updated/patched, so we can reduce the megarule to:

```
rules:
- apiGroups: [""]
  resources: ["events"]
  verbs: ["get", "list", "watch", "update", "create", "patch"]
- apiGroups: [""]
  resources: ["endpoints", "services", "pods", "nodes", "namespaces"]
  verbs: ["get", "list", "watch"]
# TODO: switch to patch services/status
# https://github.com/kubernetes/ingress-gce/blob/4918eb2f0f484f09ac9e5a975907a9b16ed2b344/pkg/neg/controller.go#L339-L342
# https://github.com/kubernetes/ingress-gce/blob/4918eb2f0f484f09ac9e5a975907a9b16ed2b344/pkg/neg/controller.go#L359-L361
- apiGroups: [""]
  resources: ["services"]
  verbs: ["update", "patch"]
```